api/server/router/image/image_routes.go

```diff
@@ -342,3 +349,106 @@ func (s *imageRouter) postImagesPrune(ctx context.Context, w http.ResponseWriter
+	if err := httputils.ParseForm(r); err != nil {
+		return err
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	var (
+		config          = &types.AuthConfig{}
+		authEncoded     = r.Header.Get("X-Registry-Auth")
+		distrepo        distribution.Repository
+		manifestInspect registrytypes.ManifestInspect
+		mnfstJSON       []byte
+		configJSON      []byte
+	)
+
+	if authEncoded != "" {
+		authJSON := base64.NewDecoder(base64.URLEncoding, strings.NewReader(authEncoded))
+		if err := json.NewDecoder(authJSON).Decode(&config); err != nil {
+			// for a search it is not an error if no auth was given
+			// to increase compatibility with the existing api it is defaulting to be empty
+			config = &types.AuthConfig{}
+		}
+	}
+
+	image := vars["name"]
+
+	ref, err := reference.ParseAnyReference(image)
+	if err != nil {
+		return err
+	}
+	namedRef, ok := ref.(reference.Named)
+	if !ok {
+		if _, ok := ref.(reference.Digested); ok {
+			// full image ID
+			return errors.Errorf("no manifest found for full image ID")
+		}
+		return errors.Errorf("unknown image reference format: %s", image)
+	}
+
+	if canonicalRef, ok := namedRef.(reference.Canonical); !ok {
+		namedRef = reference.TagNameOnly(namedRef)
+
+		taggedRef, ok := namedRef.(reference.NamedTagged)
+		if !ok {
+			return errors.Errorf("image reference not tagged: %s", image)
+		}
+
+		distrepo, _, err = s.backend.GetRepository(ctx, taggedRef, config)
+		if err != nil {
+			return err
+		}
+		dscrptr, err := distrepo.Tags(ctx).Get(ctx, taggedRef.Tag())
+		if err != nil {
+			return err
+		}
+		manifestInspect.Digest = dscrptr.Digest
+	} else {
+		manifestInspect.Digest = canonicalRef.Digest()
+		distrepo, _, err = s.backend.GetRepository(ctx, canonicalRef, config)
+		if err != nil {
+			return err
+		}
+	}
+	// at this point, we have a digest, so we can retrieve the manifest
+
+	mnfstsrvc, _ := distrepo.Manifests(ctx)
+	mnfst, _ := mnfstsrvc.Get(ctx, manifestInspect.Digest)
+	blobsrvc := distrepo.Blobs(ctx)
+
+	// retrieve manifest and config JSON depending on the type of manifest
+	switch mnfstObj := mnfst.(type) {
+	case *manifestlist.DeserializedManifestList:
+		manifestInspect.ManifestType = manifestlist.MediaTypeManifestList
+		mnfstJSON, err = mnfstObj.MarshalJSON()
+		if err != nil {
+			return err
+		}
+		// configJSON might also be set here
+	case *schema2.DeserializedManifest:
+		manifestInspect.ManifestType = schema2.MediaTypeManifest
+		mnfstJSON, err = mnfstObj.MarshalJSON()
+		if err != nil {
+			return err
+		}
+		configJSON, err = blobsrvc.Get(ctx, mnfstObj.Config.Digest)
+		if err != nil {
+			return err
+		}
+	case *schema1.SignedManifest:
+		manifestInspect.ManifestType = schema1.MediaTypeSignedManifest
+		mnfstJSON, err = mnfstObj.MarshalJSON()
+		// no configJSON to set here
+	}
+
+	mnfstObj := json.RawMessage(mnfstJSON)
+	configObj := json.RawMessage(configJSON)
+	manifestInspect.Manifest = &mnfstObj
+	manifestInspect.ConfigJSON = &configObj
```

Comment:
Let's move the lines related to `ConfigJSON` into the `*schema2.DeserializedManifest` case so that `manifestInspect.ConfigJSON` remains `nil` when other types of manifests are returned.